## 신혼부부 인테리어 패턴 분석하기

### 데이터 불러오기

In [63]:
import pandas as pd
import matplotlib.pyplot as plt

In [64]:
cus =  pd.read_csv('housewarming2.csv')
prd = pd.read_csv('hou_prd_details.csv')
prd.drop(columns='Unnamed: 0', inplace=True)

In [65]:
cus

id         작성일                              제목          가족형태     공간  \
0        0  2022-10-07  서재형 거실부터 베란다 아뜰리에까지! 센스만점 오베네홈      아기가 있는 집    아파트   
1        1  2022-10-07  구축 2Bay 리모델링의 바이블, 모던한 32평 신혼집          신혼부부    아파트   
2        2  2022-10-07    취향을 읽고 쓰는 비밀 서재처럼- 푸릇푸릇 3평 방  부모님과 함께 사는 집     기타   
3        3  2022-10-07   손 뻗으면 닿을 듯한 녹음🌳 싱그러운 저층집 리모델링          신혼부부    아파트   
4        4  2022-10-07  아늑하고 편안한 무드로- 빈티지 한 스푼 더한 내추럴홈          신혼부부    아파트   
...    ...         ...                             ...           ...    ...   
4895  4895  2019-10-27          25년 된 신혼집 반셀프 인테리어 도전기          신혼부부    아파트   
4896  4896  2019-10-26  애정 가득 신혼집 : 작은 평수 아파트도 넓고 쾌적하게          신혼부부    아파트   
4897  4897  2019-10-26              제주, 그리고 우리 셋 삶의 공간      아기가 있는 집  빌라&연립   
4898  4898  2019-10-26  지금의 행복을 택하는 건 꽤 의미있는 투자일지도 몰라요      아기가 있는 집    아파트   
4899  4899  2019-10-26         꿈꾸던 신혼집을 만들어가다 #게다가_2층집          신혼부부     기타   

      평수구간              스타일                              세부공사     분야      작업  \
0       35              내추럴                                 0   리모델링     전문가   
1       35               모던  주방리모델링, 조명시공, 중문, 가벽&파티션, 슬라이딩도어   리모델링     전문가   
2        5              내추럴                                 0  홈스타일링  셀프•DIY   
3       35               모던                  주방리모델링, 조명시공, 중문   리모델링     전문가   
4       55              내추럴                        주방리모델링, 중문   리모델링     전문가   
...    ...              ...                               ...    ...     ...   
4895    35  모던, 미니멀&심플, 북유럽  주방리모델링, 조명시공, 중문, 가벽&파티션, 슬라이딩도어   리모델링     반셀프   
4896    25       모던, 미니멀&심플                          조명시공, 중문   부분공사     반셀프   
4897    35           미니멀&심플                              조명시공  홈스타일링     전문가   
4898    25      미니멀&심플, 내추럴                주방리모델링, 중문, 슬라이딩도어   리모델링     전문가   
4899    35               모던                                 0   리모델링     전문가   

        예산     조회   스크랩  좋아요   댓글  \
0        0  10513   136   57   15   
1     4200   5610    75   26   12   
2        0   4214   201   60   15   
3     5500  37375   643  169  122   
4     9500   3760    73   13    9   
...    ...    ...   ...  ...  ...   
4895  3000  82539  2865  723   67   
4896   500  45480   713  236   30   
4897     0  39918   350  201   22   
4898     0  94423  2529  980   88   
4899     0  53937   958  507   57   

                                                    url  
0     https://ohou.se/projects/126917/detail?affect_...  
1     https://ohou.se/projects/126600/detail?affect_...  
2     https://ohou.se/projects/123056/detail?affect_...  
3     https://ohou.se/projects/116945/detail?affect_...  
4     https://ohou.se/projects/127436/detail?affect_...  
...                                                 ...  
4895  https://ohou.se/projects/17105/detail?affect_t...  
4896  https://ohou.se/projects/16242/detail?affect_t...  
4897  https://ohou.se/projects/15965/detail?affect_t...  
4898  https://ohou.se/projects/7660/detail?affect_ty...  
4899  https://ohou.se/projects/7490/detail?affect_ty...  

[4900 rows x 16 columns]

In [31]:
prd

id                                        상품url     브랜드명  \
0          0   https://ohou.se/productions/538782/selling      에어론   
1          0   https://ohou.se/productions/803775/selling     베네베네   
2          0  https://ohou.se/productions/1068049/selling    플렌스테드   
3          0   https://ohou.se/productions/580038/selling      잠솔솔   
4          0   https://ohou.se/productions/961600/selling       리스   
...      ...                                          ...      ...   
243014  4691   https://ohou.se/productions/224590/selling    아키드리머   
243015  4691   https://ohou.se/productions/224582/selling    아키드리머   
243016  4691   https://ohou.se/productions/224584/selling  럭키보이선데이   
243017  4691    https://ohou.se/productions/11729/selling      이케아   
243018  4691   https://ohou.se/productions/153822/selling      이케아   

                                            상품명  \
0       리버서블 오피스 데스크 책상/테이블 600~2400mm(2colors)   
1                      벤트라이크 멀티플 접이식 세발자전거+안전가드   
2         FLENSTED MOBILES 모빌 - 천사 Angel mobile   
3                     경추목 메모리폼 숙면 꿀잠 수면 베개_항균탠샐   
4                        리스 클래식 법랑 양수냄비 22cm(H)   
...                                         ...   
243014                      악세스시리즈01 한정판 아트포스터    
243015                          [해외]Sand Rug 01   
243016                               Pale Nulle   
243017                        FROSTA 스툴 3colors   
243018                       EKRAR 에크라르 스탠드 옷걸이   

                                     카테고리      상품가격     리뷰수     스크랩수  스타일링샷수  
0       ['가구', '테이블·식탁·책상', '책상', '일반책상']   39800.0  2891.0  57140.0  2113.0  
1           ['유아·아동', '완구·교구', '공간·대형완구']   94800.0     1.0     60.0     1.0  
2           ['데코·식물', '인테리어소품', '모빌·가랜드']   37400.0     NaN     81.0     0.0  
3               ['패브릭', '베개·베개커버', '폼베개']   38900.0    32.0     78.0    13.0  
4         ['주방용품', '냄비·프라이팬·솥', '냄비·뚝배기']  135000.0     1.0     89.0     1.0  
...                                   ...       ...     ...      ...     ...  
243014            ['데코·식물', '홈갤러리', '액자']       NaN     NaN     32.0     3.0  
243015       ['패브릭', '러그·카페트', '라탄·수초러그']       NaN     NaN     26.0     8.0  
243016        ['데코·식물', '인테리어소품', '장식소품']       NaN     NaN     12.0     7.0  
243017              ['가구', '의자', '스툴·벤치']       NaN     NaN  17597.0  1994.0  
243018     ['가구', '행거·옷장', '행거', '스탠드행거']       NaN     NaN    639.0    24.0  

[243019 rows x 9 columns]

### 데이터 조인

In [66]:
master = pd.merge(cus, prd, on = 'id', how = 'inner')
master.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 243019 entries, 0 to 243018
Data columns (total 24 columns):
 #   Column  Non-Null Count   Dtype  
---  ------  --------------   -----  
 0   id      243019 non-null  int64  
 1   작성일     243019 non-null  object 
 2   제목      243019 non-null  object 
 3   가족형태    243019 non-null  object 
 4   공간      243019 non-null  object 
 5   평수구간    243019 non-null  int64  
 6   스타일     243019 non-null  object 
 7   세부공사    243019 non-null  object 
 8   분야      243019 non-null  object 
 9   작업      243019 non-null  object 
 10  예산      243019 non-null  int64  
 11  조회      243019 non-null  int64  
 12  스크랩     243019 non-null  int64  
 13  좋아요     243019 non-null  int64  
 14  댓글      243019 non-null  int64  
 15  url     243019 non-null  object 
 16  상품url   243019 non-null  object 
 17  브랜드명    243019 non-null  object 
 18  상품명     243019 non-null  object 
 19  카테고리    243019 non-null  object 
 20  상품가격    164076 non-null  float64
 21  리뷰수     15

In [33]:
master['작성일'] = pd.to_datetime(master['작성일'])
master.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 243019 entries, 0 to 243018
Data columns (total 24 columns):
 #   Column  Non-Null Count   Dtype         
---  ------  --------------   -----         
 0   id      243019 non-null  int64         
 1   작성일     243019 non-null  datetime64[ns]
 2   제목      243019 non-null  object        
 3   가족형태    243019 non-null  object        
 4   공간      243019 non-null  object        
 5   평수구간    243019 non-null  int64         
 6   스타일     243019 non-null  object        
 7   세부공사    243019 non-null  object        
 8   분야      243019 non-null  object        
 9   작업      243019 non-null  object        
 10  예산      243019 non-null  int64         
 11  조회      243019 non-null  int64         
 12  스크랩     243019 non-null  int64         
 13  좋아요     243019 non-null  int64         
 14  댓글      243019 non-null  int64         
 15  url     243019 non-null  object        
 16  상품url   243019 non-null  object        
 17  브랜드명    243019 non-null  obje

### 분석에 사용할 칼럼 가져오기

In [67]:
df = master[['id', '작성일', '제목', '가족형태', '공간', '평수구간', '스타일', '예산', '브랜드명', '상품명', '카테고리', '상품가격']]
df

id         작성일                              제목      가족형태    공간  \
0          0  2022-10-07  서재형 거실부터 베란다 아뜰리에까지! 센스만점 오베네홈  아기가 있는 집   아파트   
1          0  2022-10-07  서재형 거실부터 베란다 아뜰리에까지! 센스만점 오베네홈  아기가 있는 집   아파트   
2          0  2022-10-07  서재형 거실부터 베란다 아뜰리에까지! 센스만점 오베네홈  아기가 있는 집   아파트   
3          0  2022-10-07  서재형 거실부터 베란다 아뜰리에까지! 센스만점 오베네홈  아기가 있는 집   아파트   
4          0  2022-10-07  서재형 거실부터 베란다 아뜰리에까지! 센스만점 오베네홈  아기가 있는 집   아파트   
...      ...         ...                             ...       ...   ...   
243014  4691  2020-01-01  늘 취향을 곁에 두고 살아요ㅣ어느 디자이너의 런던 살이  아기가 있는 집  단독주택   
243015  4691  2020-01-01  늘 취향을 곁에 두고 살아요ㅣ어느 디자이너의 런던 살이  아기가 있는 집  단독주택   
243016  4691  2020-01-01  늘 취향을 곁에 두고 살아요ㅣ어느 디자이너의 런던 살이  아기가 있는 집  단독주택   
243017  4691  2020-01-01  늘 취향을 곁에 두고 살아요ㅣ어느 디자이너의 런던 살이  아기가 있는 집  단독주택   
243018  4691  2020-01-01  늘 취향을 곁에 두고 살아요ㅣ어느 디자이너의 런던 살이  아기가 있는 집  단독주택   

        평수구간       스타일  예산     브랜드명                                      상품명  \
0         35       내추럴   0      에어론  리버서블 오피스 데스크 책상/테이블 600~2400mm(2colors)   
1         35       내추럴   0     베네베네                 벤트라이크 멀티플 접이식 세발자전거+안전가드   
2         35       내추럴   0    플렌스테드    FLENSTED MOBILES 모빌 - 천사 Angel mobile   
3         35       내추럴   0      잠솔솔                경추목 메모리폼 숙면 꿀잠 수면 베개_항균탠샐   
4         35       내추럴   0       리스                   리스 클래식 법랑 양수냄비 22cm(H)   
...      ...       ...  ..      ...                                      ...   
243014    25  유니크&믹스매치   0    아키드리머                      악세스시리즈01 한정판 아트포스터    
243015    25  유니크&믹스매치   0    아키드리머                          [해외]Sand Rug 01   
243016    25  유니크&믹스매치   0  럭키보이선데이                               Pale Nulle   
243017    25  유니크&믹스매치   0      이케아                        FROSTA 스툴 3colors   
243018    25  유니크&믹스매치   0      이케아                       EKRAR 에크라르 스탠드 옷걸이   

                                     카테고리      상품가격  
0       ['가구', '테이블·식탁·책상', '책상', '일반책상']   39800.0  
1           ['유아·아동', '완구·교구', '공간·대형완구']   94800.0  
2           ['데코·식물', '인테리어소품', '모빌·가랜드']   37400.0  
3               ['패브릭', '베개·베개커버', '폼베개']   38900.0  
4         ['주방용품', '냄비·프라이팬·솥', '냄비·뚝배기']  135000.0  
...                                   ...       ...  
243014            ['데코·식물', '홈갤러리', '액자']       NaN  
243015       ['패브릭', '러그·카페트', '라탄·수초러그']       NaN  
243016        ['데코·식물', '인테리어소품', '장식소품']       NaN  
243017              ['가구', '의자', '스툴·벤치']       NaN  
243018     ['가구', '행거·옷장', '행거', '스탠드행거']       NaN  

[243019 rows x 12 columns]

### 신혼부부 구매 패턴 분석

In [35]:
# 신혼부부 데이터만 가져오기
df = df[df['가족형태'].str.contains('신혼부부')]
df

id        작성일                              제목  가족형태   공간  평수구간  \
100        1 2022-10-07  구축 2Bay 리모델링의 바이블, 모던한 32평 신혼집  신혼부부  아파트    35   
101        1 2022-10-07  구축 2Bay 리모델링의 바이블, 모던한 32평 신혼집  신혼부부  아파트    35   
102        1 2022-10-07  구축 2Bay 리모델링의 바이블, 모던한 32평 신혼집  신혼부부  아파트    35   
103        1 2022-10-07  구축 2Bay 리모델링의 바이블, 모던한 32평 신혼집  신혼부부  아파트    35   
104        1 2022-10-07  구축 2Bay 리모델링의 바이블, 모던한 32평 신혼집  신혼부부  아파트    35   
...      ...        ...                             ...   ...  ...   ...   
242920  4688 2020-01-02    소중한 공간을 소중하게 채워나가는 소중한 내 인생   신혼부부  아파트    25   
242921  4688 2020-01-02    소중한 공간을 소중하게 채워나가는 소중한 내 인생   신혼부부  아파트    25   
242922  4688 2020-01-02    소중한 공간을 소중하게 채워나가는 소중한 내 인생   신혼부부  아파트    25   
242923  4688 2020-01-02    소중한 공간을 소중하게 채워나가는 소중한 내 인생   신혼부부  아파트    25   
242924  4688 2020-01-02    소중한 공간을 소중하게 채워나가는 소중한 내 인생   신혼부부  아파트    25   

               스타일    예산   브랜드명                                           상품명  \
100             모던  4200   더죤테크                         제시단101 원홀세면수전 (색상 6종)   
101             모던  4200  모던하우스  ON 높이조절 셀프맞춤 100% 빛차단 반창문 암막커튼 140x180x1P화이트   
102             모던  4200   어반테고       TET-마시멜로우 MARSH MALLOW 테라조 타일/인테리어 패턴타일   
103             모던  4200  모던하우스               먼지없는 100% 암막커튼 베이지 220x240 2개세트   
104             모던  4200  반석스포츠         비스펙 치닝디핑 블랙 턱걸이 철봉 평행봉 풀업운동 헬스기구 운동기구   
...            ...   ...    ...                                           ...   
242920  모던, 미니멀&심플   100    리센트                         초대형 전신거울 6종 홈피트니스 피팅룸   
242921  모던, 미니멀&심플   100    에라토                                노르딕 3단 도자기 수저통   
242922  모던, 미니멀&심플   100    마이룸                      루맥스 미니 슬라이딩 리빙박스 4colors   
242923  모던, 미니멀&심플   100   창신리빙                           내츄럴 서랍형 리빙박스 투명수납함    
242924  모던, 미니멀&심플   100    시예로            아일랜드식탁의자 바스툴 CR-B181 카우보이 3 color    

                                     카테고리      상품가격  
100           ['공구·DIY', '수도', '수전·수도용품']  166000.0  
101             ['패브릭', '커튼·부자재', '암막커튼']   23300.0  
102                  ['공구·DIY', '타일·파벽돌']   42900.0  
103             ['패브릭', '커튼·부자재', '암막커튼']   89100.0  
104                  ['실내운동', '근력', '철봉']   95000.0  
...                                   ...       ...  
242920               ['가구', '거울', '전신거울']       NaN  
242921         ['주방용품', '주방수납·정리', '수저통']       NaN  
242922  ['수납·정리', '리빙박스·바구니', '플라스틱리빙박스']    2300.0  
242923     ['수납·정리', '리빙박스·바구니', '기타수납함']       NaN  
242924                ['가구', '의자', '바체어']       NaN  

[106298 rows x 12 columns]

In [36]:
df['가족형태'].value_counts()

신혼부부                  102942
신혼부부, 아기가 있는 집          1717
신혼부부, 기타                1463
신혼부부, 부모님과 함께 사는 집        65
신혼부부, 아기가 있는 집, 기타        64
싱글라이프, 신혼부부, 기타           47
Name: 가족형태, dtype: int64

In [37]:
# 신혼부부 총 유저 수
print(len(df['id'].unique()))

1909


In [38]:
# 유저별 몇 개의 상품을 구매하나?
customer_unique_transaction_series = df.groupby('id')['상품명'].nunique()
customer_unique_transaction_series.describe()

count    1909.000000
mean       55.618125
std        26.109335
min         5.000000
25%        37.000000
50%        53.000000
75%        71.000000
max       191.000000
Name: 상품명, dtype: float64

인테리어를 할 때 유저당 평균 55개의 상품 구입, 최소5개 - 최대191개

In [39]:
# 유저별 아이템 카테고리 종류 개수를 탐색
customer_unique_item_series = df.groupby('id')['카테고리'].nunique()
customer_unique_item_series.describe()

count    1909.000000
mean       44.427449
std        19.110092
min         3.000000
25%        31.000000
50%        43.000000
75%        57.000000
max       130.000000
Name: 카테고리, dtype: float64

인테리어를 할 때 유저당 평균 44개의 카테고리에서 구입, 최소3개 - 최대130개   

In [40]:
# 가장 구매가 많은 카테고리 top 10 탐색
df['카테고리'].value_counts()

['가구', '의자', '인테리어의자']                     2838
['데코·식물', '플라워·식물', '화병·화분', '화병']         2723
['조명', '단스탠드']                             1642
['주방용품', '그릇·홈세트', '접시·플레이트', '일반접시']      1641
['가구', '테이블·식탁·책상', '식탁', '식탁·입식테이블']      1575
                                           ... 
['캠핑용품', '캠핑부품·공구', '기타부품']                   1
['가전', '주방가전', '수비드머신·진공포장기', '수비드머신']        1
['캠핑용품', '침낭·매트·해먹', '해먹']                    1
['가전', '생활·건강가전', '다리미·보풀제거기', '보풀제거기']       1
['생필품', '스킨케어', '페이셜케어', '앰플·오일']             1
Name: 카테고리, Length: 877, dtype: int64

선물하기 좋은 제품 카페고리 선정: 가구(의자) > 데코(화분) > 조명 > 주방용품. 가구는 선물하기 어렵고 데코와 큰 차이가 나지 않음.   
가구는 제외하거나 소가구만 보는 것도 방법

In [41]:
# 가장 구매가 많은 상품 top 10
item_count = df['상품명'].value_counts()[:10]
for idx, (val, cnt) in enumerate(item_count.iteritems(), 1):
    print("Top", idx, ":", val, cnt)

Top 1 : [최종가29.9만][1만쿠폰] 발뮤다 뉴토스터 3.0 + (9만원 상당 정품 사은품) 292
Top 2 : [최종가15.9만][1만쿠폰] 발뮤다 더 팟 전기포트 전기주전자 194
Top 3 : 독일토분 표준형 테라코타 빈티지 화분 182
Top 4 : 루씨에어 실링팬 레이더 (한국공식수입원_A/S가능) 133
Top 5 : 인테리어 우드액자 프레임 8종  A1 A2 A3 A4   132
Top 6 : [해외] Aesop 레저렉션 아로마틱 핸드워시 500ml 2개세트 131
Top 7 : 베가 폴라 투명 접이식의자 17colors(대량구매 추가할인) 114
Top 8 : [무료배송] 음쓰통계의 샤넬! 18-8 풀스텐 음식물쓰레기통 3L 96
Top 9 : LED T5 간접조명_6colors 95
Top 10 : 밀라노 1500 북선반 5colors 91


선물하기 좋은 상품 선정: 주방가전, 반려식물, 액자 위주의 선물기획

In [42]:
# 공간별로 상품에 얼마를 지출하나?
df2 = df.dropna(subset=['공간','상품가격'])  # 상품가격이 null인 행 제거
df2.groupby('공간')['상품가격'].mean().sort_values(ascending = False)

공간
아파트        221264.250808
단독주택       196355.517739
원룸&오피스텔    165156.131760
빌라&연립      163737.696560
사무공간       153466.666667
기타         114540.338592
Name: 상품가격, dtype: float64

아파트 > 단독주택 > 원룸&오피스텔 > 빌라&연립 > 사무공간 > 기타
- 아파트가 평균 지출액이 높은데 아파트에 거주하는 신혼부부가 많고 처음 입주할 때 전체 인테리어를 많이 하기 때문이라고 판단